app/src/main/java/com/capstone/weatherapp/CityListFragment.kt

```diff
@@ -41,9 +45,40 @@ class CityListFragment : Fragment() {
         rvCities.layoutManager = LinearLayoutManager(context)
 
         viewModel.cityList.observe(viewLifecycleOwner) {
-            cityListAdapter.updateData(it)
+            when(it) {
+                is State.Success -> showSuccessState(it.result)
+                is State.Loading -> showLoadingState()
+                is State.Error -> showErrorState(it.message)
+            }
         }
         viewModel.getCityListFromRepo()
     }
 
+    fun showLoadingState() {
+        binding.rvCities.visibility = View.GONE
+        binding.stateLayout.visibility = View.VISIBLE
+        binding.detailsState.text = ""
+        Glide.with(this)
+            .load(R.drawable.loading)
+            .centerCrop()
+            .into(binding.detailsStateIcon)
+    }
+
+    fun showErrorState(message: String) {
+        binding.rvCities.visibility = View.GONE
+        binding.stateLayout.visibility = View.VISIBLE
+        binding.detailsState.text = message
```

Comment:
We don't want the error message on the error screen to be the actual error, the user doesn't need to know what the actual error is. Take another look at the [this](https://drive.google.com/file/d/1C-HMBdCgtcowO00H_HBSVbZAhCPLWLNh/view) and use the generic error message